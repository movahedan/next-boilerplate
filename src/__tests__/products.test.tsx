import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { act } from 'react-dom/test-utils';
import { create } from 'react-test-renderer';

import { mockUseRouter } from '__tests__/__helpers__/router.test';
import ProductsPage, { getStaticProps } from 'pages/products/index';

import type { InferGetStaticPropsType } from 'next';

mockUseRouter();

describe('<ProductsPage>', () => {
	const ids = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
	const defaultProps: InferGetStaticPropsType<typeof getStaticProps> = {
		data: {
			ids,
		},
	};

	it('should match snapshot', async () => {
		const component = create(<ProductsPage {...defaultProps} />).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(<ProductsPage {...defaultProps} />);

		let results;
		await act(async () => {
			results = await axe(container);
		});
		expect(results).toHaveNoViolations();
	});

	it('should have getStaticProps', async () => {
		const results = await getStaticProps({});

		expect(results).toEqual({
			props: {
				data: {
					ids: [...ids, ids.length + 1],
				},
			},
			revalidate: 1,
		});
	});
});
