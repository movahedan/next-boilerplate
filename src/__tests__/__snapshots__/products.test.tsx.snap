// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsPage> should match snapshot 1`] = `
<div>
  <h1>
    Products page
  </h1>
  <ul>
    <li>
      <a
        href="/products/1"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        1
      </a>
    </li>
    <li>
      <a
        href="/products/2"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        2
      </a>
    </li>
    <li>
      <a
        href="/products/3"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        3
      </a>
    </li>
    <li>
      <a
        href="/products/4"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        4
      </a>
    </li>
    <li>
      <a
        href="/products/5"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        5
      </a>
    </li>
    <li>
      <a
        href="/products/6"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        6
      </a>
    </li>
    <li>
      <a
        href="/products/7"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        7
      </a>
    </li>
    <li>
      <a
        href="/products/8"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        8
      </a>
    </li>
    <li>
      <a
        href="/products/9"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        9
      </a>
    </li>
    <li>
      <a
        href="/products/10"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        10
      </a>
    </li>
  </ul>
</div>
`;
